apps/backend/src/ws.ts

```diff
@@ -0,0 +1,117 @@
+import { WebSocket } from 'ws';
+import {
+  WebSocketRequestTypes,
+  WebSocketRequest,
+  WebSocketRequestSchema,
+  WebSocketResponseTypes,
+  WebSocketResponse,
+  WebSocketErrorPayload,
+  MapRequestToResponse,
+  errorToString
+} from "@types";
+import { Controller } from "@/lib/controller";
+
+import { wsServer } from "@/index";
+
+const controller = new Controller();
+
+const clientsSockets: Record<string, WebSocket> = {};
```

Comment:
What are the implications of this being global, it just resets every server restart? Is there worry of this ever becoming too large?